WordPress/src/main/java/org/wordpress/android/util/LocaleManager.java

```diff
@@ -0,0 +1,279 @@
+package org.wordpress.android.util;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.res.Configuration;
+import android.content.res.Resources;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.support.annotation.NonNull;
+import android.text.TextUtils;
+import android.util.Pair;
+
+import java.text.Collator;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import javax.annotation.Nullable;
+
+/**
+ * Helper class for working with localized strings. Ensures updates to the users
+ * selected language is properly saved and resources appropriately updated for the
+ * android version.
+ */
+public class LocaleManager {
+    /**
+     * Length of a {@link String} (representing a language code) when there is no region included.
+     * For example: "en" contains no region, "en_US" contains a region (US)
+     * <p>
+     * Used to parse a language code {@link String} when creating a {@link Locale}.
+     */
+    private static final int NO_REGION_LANG_CODE_LEN = 2;
+
+    /**
+     * Index of a language code {@link String} where the region code begins. The language code
+     * format is cc_rr, where cc is the country code (e.g. en, es, az) and rr is the region code
+     * (e.g. us, au, gb).
+     */
+    private static final int REGION_SUBSTRING_INDEX = 3;
+
+    /**
+     * Key used for saving the language selection to shared preferences.
+     */
+    private static final String LANGUAGE_KEY = "language-pref";
+
+    /**
+     * Activate the locale associated with the provided context.
+     * @param context The current context.
+     */
+    public static Context setLocale(Context context) {
+        return updateResources(context, getLanguage(context));
+    }
+
+    /**
+     * Change the active locale to the language provided. Save the updated language
+     * settings to sharedPreferences.
+     * @param context The current context
+     * @param language The 2-letter language code (example "en") to switch to
+     */
+    public static Context setNewLocale(Context context, String language) {
+        Locale newLocale = languageLocale(language);
+
+        if (Locale.getDefault().toString().equals(newLocale.toString())) {
```

Comment:
I think this change was missed https://github.com/wordpress-mobile/WordPress-Android/pull/7413#discussion_r173547994